### PR TITLE
Droth 79 auto correction objects when geometry updated

### PR DESCRIPTION
**Pull request by marquesfm**

### CODE REVIEW

**Comment by halied** [comment]
Ok

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/PedestrianCrossingService.scala [hunk 1]
```diff
@@ -39,5 +40,88 @@ class PedestrianCrossingService(val roadLinkService: RoadLinkService) extends Po
...
+      assetsBeforeUpdate.foreach { asset =>
+        if (asset.asset.floating != asset.persistedFloating) {
+          updateFloating(asset.asset.id, asset.asset.floating, asset.floatingReason)
+        }
+      }
+      assetsBeforeUpdate.map(_.asset)
+    }
+  }
+
+  override def getByMunicipality(municipalityCode: Int): Seq[PersistedAsset] = {
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/PedestrianCrossingService.scala]: It seems like there is a lot of duplicated code we can have a method on the base class that has a function as a parameter, and then reuse it.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/PedestrianCrossingService.scala [hunk 2]
```diff
@@ -39,5 +40,88 @@ class PedestrianCrossingService(val roadLinkService: RoadLinkService) extends Po
     }
     id
   }
+
+  override def getByBoundingBox(user: User, bounds: BoundingRectangle): Seq[PersistedAsset] = {
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/PedestrianCrossingService.scala]: It seems like there is a lot of duplicated code we can have a method on the base class that has a function as a parameter, and then reuse it.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/ObstacleService.scala [hunk 3]
```diff
@@ -34,14 +38,99 @@ class ObstacleService(val roadLinkService: RoadLinkService) extends PointAssetOp
...
+        if (asset.asset.floating != asset.persistedFloating) {
+          updateFloating(asset.asset.id, asset.asset.floating, asset.floatingReason)
+        }
+      }
+
+      assetsBeforeUpdate.map(_.asset)
+    }
+  }
+
+  override def getByMunicipality(municipalityCode: Int): Seq[PersistedAsset] = {
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/ObstacleService.scala]: It seems like there is a lot of duplicated code we can have a method on the base class that has a function as a parameter, and then reuse it.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/ObstacleService.scala [hunk 4]
```diff
@@ -34,14 +38,99 @@ class ObstacleService(val roadLinkService: RoadLinkService) extends PointAssetOp
     id
   }
 
+  override def getByBoundingBox(user: User, bounds: BoundingRectangle): Seq[PersistedAsset] = {
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/ObstacleService.scala]: It seems like there is a lot of duplicated code we can have a method on the base class that has a function as a parameter, and then reuse it.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RailwayCrossingService.scala [hunk 5]
```diff
@@ -19,6 +22,91 @@ class RailwayCrossingService(val roadLinkService: RoadLinkService) extends Point
     persistedAsset.copy(floating = floating)
   }
 
+  override def getByBoundingBox(user: User, bounds: BoundingRectangle): Seq[PersistedAsset] = {
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RailwayCrossingService.scala]: It seems like there is a lot of duplicated code we can have a method on the base class that has a function as a parameter, and then reuse it.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RailwayCrossingService.scala [hunk 6]
```diff
@@ -19,6 +22,91 @@ class RailwayCrossingService(val roadLinkService: RoadLinkService) extends Point
...
+        if (asset.asset.floating != asset.persistedFloating) {
+          updateFloating(asset.asset.id, asset.asset.floating, asset.floatingReason)
+        }
+      }
+      assetsBeforeUpdate.map(_.asset)
+    }
+  }
+
+
+  override def getByMunicipality(municipalityCode: Int): Seq[PersistedAsset] = {
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/RailwayCrossingService.scala]: It seems like there is a lot of duplicated code we can have a method on the base class that has a function as a parameter, and then reuse it.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/TrafficLightService.scala [hunk 7]
```diff
@@ -34,4 +37,87 @@ class TrafficLightService(val roadLinkService: RoadLinkService) extends PointAss
       OracleTrafficLightDao.create(TrafficLightToBePersisted(asset.linkId, asset.lon, asset.lat, mValue, municipality, username), username)
     }
   }
+
+  override def getByBoundingBox(user: User, bounds: BoundingRectangle): Seq[PersistedAsset] = {
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/TrafficLightService.scala]: It seems like there is a lot of duplicated code we can have a method on the base class that has a function as a parameter, and then reuse it.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/TrafficLightService.scala [hunk 8]
```diff
@@ -34,4 +37,87 @@ class TrafficLightService(val roadLinkService: RoadLinkService) extends PointAss
...
+      assetsBeforeUpdate.foreach { asset =>
+        if (asset.asset.floating != asset.persistedFloating) {
+          updateFloating(asset.asset.id, asset.asset.floating, asset.floatingReason)
+        }
+      }
+      assetsBeforeUpdate.map(_.asset)
+    }
+  }
+
+  override def getByMunicipality(municipalityCode: Int): Seq[PersistedAsset] = {
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/TrafficLightService.scala]: It seems like there is a lot of duplicated code we can have a method on the base class that has a function as a parameter, and then reuse it.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/PointAssetFiller.scala [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+            case _ => None
+          }
+        case _ => None
+      }.head
+    } else {
+      None
+    }
+  }
+
+  def correctOnlyGeometry(asset: PersistedPointAsset, roadLinks: Seq[RoadLink], changeInfos: Seq[ChangeInfo]): Option[AssetAdjustment] = {
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/PointAssetFiller.scala]: I belive we can the logic in separeted methods to be easer to read the code.

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/PointAssetFiller.scala [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+import org.joda.time.DateTime
+
+object PointAssetFiller {
+  case class testPersistedPointAsset(id: Long, lon: Double, lat: Double, municipalityCode: Int, linkId: Long, mValue: Double, floating: Boolean) extends PersistedPointAsset
+
+
+  case class AssetAdjustment(assetId: Long, lon: Double, lat: Double, linkId: Long, mValue: Double, floating: Boolean)
+  private val MaxDistanceDiffAllowed = 3.0
+
+  def correctRoadLinkAndGeometry(asset: PersistedPointAsset , roadLinks: Seq[RoadLink], changeInfos: Seq[ChangeInfo]) : Option [AssetAdjustment] = {
```
Comment (leirasf) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/PointAssetFiller.scala]: This is  not possible to do only with the mValues?

**Comment by marquesfm** [comment]
All ok for me

**Pull request merged**